src/components/drawing/symbol_defs.js

```diff
@@ -10,544 +11,798 @@ var d3 = require('@plotly/d3');
         n: 40,
-        f: function(r) {
-            var y = d3.round(r * 1.2, 2);
-            var x0 = d3.round(r * 1.6, 2);
-            var x1 = d3.round(r * 0.8, 2);
-            return 'M-' + x1 + ',' + y + 'L0,0M-' + x1 + ',-' + y + 'L0,0M' + x0 + ',0L0,0';
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var y = round(r * 1.2, 2);
+            var x0 = round(r * 1.6, 2);
+            var x1 = round(r * 0.8, 2);
+            return align(angle, standoff, 'M-' + x1 + ',' + y + 'L0,0M-' + x1 + ',-' + y + 'L0,0M' + x0 + ',0L0,0');
         },
         needLine: true,
         noDot: true,
         noFill: true
     },
     'line-ew': {
         n: 41,
-        f: function(r) {
-            var rc = d3.round(r * 1.4, 2);
-            return 'M' + rc + ',0H-' + rc;
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rc = round(r * 1.4, 2);
+            return align(angle, standoff, 'M' + rc + ',0H-' + rc);
         },
         needLine: true,
         noDot: true,
         noFill: true
     },
     'line-ns': {
         n: 42,
-        f: function(r) {
-            var rc = d3.round(r * 1.4, 2);
-            return 'M0,' + rc + 'V-' + rc;
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rc = round(r * 1.4, 2);
+            return align(angle, standoff, 'M0,' + rc + 'V-' + rc);
         },
         needLine: true,
         noDot: true,
         noFill: true
     },
     'line-ne': {
         n: 43,
-        f: function(r) {
-            var rx = d3.round(r, 2);
-            return 'M' + rx + ',-' + rx + 'L-' + rx + ',' + rx;
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rx = round(r, 2);
+            return align(angle, standoff, 'M' + rx + ',-' + rx + 'L-' + rx + ',' + rx);
         },
         needLine: true,
         noDot: true,
         noFill: true
     },
     'line-nw': {
         n: 44,
-        f: function(r) {
-            var rx = d3.round(r, 2);
-            return 'M' + rx + ',' + rx + 'L-' + rx + ',-' + rx;
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rx = round(r, 2);
+            return align(angle, standoff, 'M' + rx + ',' + rx + 'L-' + rx + ',-' + rx);
         },
         needLine: true,
         noDot: true,
         noFill: true
     },
     'arrow-up': {
         n: 45,
-        f: function(r) {
-            var rx = d3.round(r, 2);
-            var ry = d3.round(r * 2, 2);
-            return 'M0,0L-' + rx + ',' + ry + 'H' + rx + 'Z';
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rx = round(r, 2);
+            var ry = round(r * 2, 2);
+            return align(angle, standoff, 'M0,0L-' + rx + ',' + ry + 'H' + rx + 'Z');
```

Comment:
`arrow-up` and `arrow-bar-up` should get backoffs.

Reply:
I am not sure about this. In addition to being a minor breaking change, those markers are specifically designed to work well without backoffs. 
On the other hand we could add new markers e.g. `arrow-bar` and `arrow-bar-wide` if you'd like to have the option of bar.

Reply:
Addressed in [7052984](https://github.com/plotly/plotly.js/pull/6297/commits/7052984dc8597b674ce06d1a65d4a2b50848055e).